diff --git a/app/build.gradle b/app/build.gradle
index 246b04b4..74fc6311 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -19,9 +19,21 @@ android {
                 arguments = [eventBusIndex: 'ml.docilealligator.infinityforreddit.EventBusIndex']
             }
         }
+        buildConfigField "String", "API_CLIENT_ID", "\"$API_CLIENT_ID\""
+        buildConfigField "String", "API_REDIRECT_URI", "\"$API_REDIRECT_URI\""
+        buildConfigField "String", "CUSTOM_USER_AGENT", "\"$CUSTOM_USER_AGENT\""
+    }
+    signingConfigs {
+        release {
+            storeFile rootProject.file(RELEASE_STORE_FILE)
+            storePassword RELEASE_STORE_PSWD
+            keyAlias RELEASE_KEY_ALIAS
+            keyPassword RELEASE_KEY_PSWD
+        }
     }
     buildTypes {
         release {
+            signingConfig signingConfigs.release
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
diff --git a/app/src/main/java/ml/docilealligator/infinityforreddit/utils/APIUtils.java b/app/src/main/java/ml/docilealligator/infinityforreddit/utils/APIUtils.java
index 4cbad523..8c055f74 100644
--- a/app/src/main/java/ml/docilealligator/infinityforreddit/utils/APIUtils.java
+++ b/app/src/main/java/ml/docilealligator/infinityforreddit/utils/APIUtils.java
@@ -10,6 +10,7 @@ import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicReference;
 
+import ml.docilealligator.infinityforreddit.BuildConfig;
 import ml.docilealligator.infinityforreddit.account.Account;
 import okhttp3.MediaType;
 import okhttp3.RequestBody;
@@ -32,7 +33,7 @@ public class APIUtils {
 
     public static final String CLIENT_ID_KEY = "client_id";
     public static final String CLIENT_SECRET_KEY = "client_secret";
-    public static final String CLIENT_ID = "";
+    public static final String CLIENT_ID = BuildConfig.API_CLIENT_ID;
     public static final String IMGUR_CLIENT_ID = "Client-ID cc671794e0ab397";
     public static final String REDGIFS_CLIENT_ID = "1828d0bcc93-15ac-bde6-0005-d2ecbe8daab3";
     public static final String REDGIFS_CLIENT_SECRET = "TJBlw7jRXW65NAGgFBtgZHu97WlzRXHYybK81sZ9dLM=";
@@ -42,7 +43,7 @@ public class APIUtils {
     public static final String STATE_KEY = "state";
     public static final String STATE = "23ro8xlxvzp4asqd";
     public static final String REDIRECT_URI_KEY = "redirect_uri";
-    public static final String REDIRECT_URI = "infinity://localhost";
+    public static final String REDIRECT_URI = BuildConfig.API_REDIRECT_URI;
     public static final String DURATION_KEY = "duration";
     public static final String DURATION = "permanent";
     public static final String SCOPE_KEY = "scope";
@@ -52,8 +53,8 @@ public class APIUtils {
     public static final String AUTHORIZATION_KEY = "Authorization";
     public static final String AUTHORIZATION_BASE = "bearer ";
     public static final String USER_AGENT_KEY = "User-Agent";
-    public static final String USER_AGENT = "";
-    public static final String ANONYMOUS_USER_AGENT = "";
+    public static final String USER_AGENT = BuildConfig.CUSTOM_USER_AGENT != "" ? BuildConfig.CUSTOM_USER_AGENT : "";
+    public static final String ANONYMOUS_USER_AGENT = BuildConfig.CUSTOM_USER_AGENT != "" ? BuildConfig.CUSTOM_USER_AGENT : "";
     public static final String USERNAME_KEY = "username";
 
     public static final String GRANT_TYPE_KEY = "grant_type";
diff --git a/gradle.properties b/gradle.properties
index fb78db58..9bba295a 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -19,3 +19,14 @@ org.gradle.jvmargs=-Xmx2048m
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
+
+APP_PACKAGE=ml.docilealligator.infinityforreddit
+
+RELEASE_STORE_FILE=Infinity.jks
+RELEASE_STORE_PSWD=Infinity
+RELEASE_KEY_ALIAS=Infinity
+RELEASE_KEY_PSWD=Infinity
+
+API_CLIENT_ID=
+API_REDIRECT_URI=infinity://localhost
+CUSTOM_USER_AGENT=
